scripts/vault-ensure-token.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+token_data="$(vault token lookup -format=json)"
+vault_token_never_expire="$(jq '.data.expire_time == null' <<< "$token_data")"
+vault_token_ttl="$(jq '.data.ttl' <<< "$token_data")"
+if [[ $vault_token_never_expire == false && $vault_token_ttl -le 0 ]]; then
+    echo 'Vault token expired or invalid. Please log into vault first.'
+    exit 1
+fi
```

Comment:
For what it's worth, note that this passage only ensures there is a token present and it has not expired. It makes no attempt at ensuring that the token is actually from the correct Vault remote, or that the token has the necessary access to perform the operations.

Reply:
I think that's just fine -- this is just a "fail-early" thing for when you forgot to authenticate